rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -2232,6 +2232,41 @@ void getRetiringFederatorPublicKeyOfType(MessageCall.MsgType msgType, Activation
         }
     }
 
+    @ParameterizedTest()
+    @MethodSource("msgTypesAndActivations")
+    void getProposedFederationAddress(MessageCall.MsgType msgType, ActivationConfig activationConfig) throws VMException {
+        Transaction rskTxMock = mock(Transaction.class);
+        doReturn(true).when(rskTxMock).isLocalCallTransaction();
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        Address expectedAddress = Address.fromBase58(networkParameters, "32Bhwee9FzQbuaG29RcXpdrvYnvZeMk11M");
+        when(bridgeSupportMock.getProposedFederationAddress()).thenReturn(Optional.of(expectedAddress));
+
+        Bridge bridge = bridgeBuilder
+            .transaction(rskTxMock)
+            .activationConfig(activationConfig)
+            .bridgeSupport(bridgeSupportMock)
+            .msgType(msgType)
+            .build();
+
+        CallTransaction.Function function = BridgeMethods.GET_PROPOSED_FEDERATION_ADDRESS.getFunction();
+        byte[] data = function.encode();
+
+        if (activationConfig.isActive(ConsensusRule.RSKIP419, 0)) {
+            if (activationConfig.isActive(ConsensusRule.RSKIP417, 0) &&
+                !(msgType.equals(MessageCall.MsgType.CALL) || msgType.equals(MessageCall.MsgType.STATICCALL))) {
+                // Post arrowhead should fail for any msg type != CALL or STATIC CALL
+                assertThrows(VMException.class, () -> bridge.execute(data));
```

Comment:
Sometimes one of the problems with parameterized tests is that it might be hard to know if certain assertion line is being reached.

A test without any branching logic, without any conditional, it's clear that reaches its assertions 100% of the time.

No big deal, probably, but I think this is something to discuss with the team.

Reply:
We already discussed it here: https://github.com/rsksmart/rskj/pull/2746#discussion_r1772028833
The idea is to do this in a new PR to not overcomplicate this one.